src/contractor/contractor.cpp

```diff
@@ -713,47 +711,27 @@ Contractor::LoadEdgeExpandedGraph(const ContractorConfig &config,
             return;
 
         // Now save out the updated compressed geometries
-        std::ofstream geometry_stream(config.geometry_path, std::ios::binary);
-        if (!geometry_stream)
-        {
-            const std::string message{"Failed to open " + config.geometry_path + " for writing"};
-            throw util::exception(message + SOURCE_REF);
-        }
+        storage::io::FileWriter geometry_file(config.geometry_path,
+                                              storage::io::FileWriter::HasNoFingerprint);
+
         const unsigned number_of_indices = geometry_indices.size();
         const unsigned number_of_compressed_geometries = geometry_node_list.size();
-        geometry_stream.write(reinterpret_cast<const char *>(&number_of_indices), sizeof(unsigned));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_indices[0])),
-                              number_of_indices * sizeof(unsigned));
-        geometry_stream.write(reinterpret_cast<const char *>(&number_of_compressed_geometries),
-                              sizeof(unsigned));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_node_list[0])),
-                              number_of_compressed_geometries * sizeof(NodeID));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_fwd_weight_list[0])),
-                              number_of_compressed_geometries * sizeof(EdgeWeight));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_rev_weight_list[0])),
-                              number_of_compressed_geometries * sizeof(EdgeWeight));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_fwd_duration_list[0])),
-                              number_of_compressed_geometries * sizeof(EdgeWeight));
-        geometry_stream.write(reinterpret_cast<char *>(&(geometry_rev_duration_list[0])),
-                              number_of_compressed_geometries * sizeof(EdgeWeight));
+
+        geometry_file.WriteOne(number_of_indices);
+        geometry_file.WriteFrom(&geometry_indices[0], number_of_indices);
+
+        geometry_file.WriteOne(number_of_compressed_geometries);
+        geometry_file.WriteFrom(&geometry_node_list[0], number_of_compressed_geometries);
+        geometry_file.WriteFrom(&geometry_fwd_weight_list[0], number_of_compressed_geometries);
+        geometry_file.WriteFrom(&geometry_rev_weight_list[0], number_of_compressed_geometries);
+        geometry_file.WriteFrom(&geometry_fwd_duration_list[0], number_of_compressed_geometries);
+        geometry_file.WriteFrom(&geometry_rev_duration_list[0], number_of_compressed_geometries);
```

Comment:
Hm seeing this, I _think_ you can just use `.data()` instead of the weird `&vec[0]` trick.